Update question about types (#1)
tc39/faq: README.md

Browsers are [unwilling to break web pages](https://developer.chrome.com/blog/smooshgate/#why-dont-we-just-keep-the-existing-name-and-break-the-web) for their users, no matter how much you think the developers of those websites deserve it. However, "not breaking the web" does not mean 100% backward compatibility. Backward-incompatible changes *can* be made when the undesirable behaviour is not being relied upon, but the web is very large, which makes it both hard to confidently determine this and unlikely except for very minor or very strange behaviours.

> ### Will JavaScript ever get a type system?

Reviewer: ```suggestion
### Will JavaScript ever support static typing?
```

Reviewer 2: maybe:
```suggestion
### Will JavaScript ever get a static type system?
```
?

Reviewer 3: @ljharb's suggestion is acceptable, but we should not elaborate any further.

Reviewer 4: "static type system" SGTM. Alternatively we could use examples:

> Will JavaScript ever get a built-in type system similar to TypeScript or Flow?

[Disclaimer: FAQ answers are not endorsed by TC39][]

> No. Types that help the developer (such as those in TypeScript) are not the same types that would help the engine, so a built-in type-checking experience will always provide inferior guarantees compared to what the JavaScript ecosystem is used to. The [type annotations](https://github.com/tc39/proposal-type-annotations) proposal is exploring parsing types with no runtime type-checking, which would be left to other tools.

Reviewer 5: ```suggestion
Not a _full_ type system. Types that help the developer (such as those in TypeScript) are not the same types that would help the engine, so a built-in type-checking experience will always provide inferior guarantees compared to what the TypeScript ecosystem is used to. The [type annotations](https://github.com/tc39/proposal-type-annotations) proposal is exploring parsing types without enforcing them, leaving other tools to perform static and/or runtime checking.
```

Reviewer 3: WFM

Reviewer: to be consistent with the question, suggestion:

```diff
- _full_ type system
+ _full_ static type system
```

also can we generalize the language around statically typed supersets (Iinb4 'TS is not a superset') so we don't appear to be favoring any particular one?

Reviewer 3: I don't think it's "favoring any particular one" to use TypeScript as an example of what developers expect from a type system. Generalising the language would just make it less clear to the kind of reader we're targeting here.

Reviewer: i completely agree with you on the first point, but it could give people the wrong impression.  I think this is one reason why the annotations proposal uses language like "TypeScript, Flow, and other static typing supersets of JavaScript".

Reviewer: More importantly, @littledan what do you think about the suggested change here?

Reviewer 2: It's not correct - it won't have *any* kind of static type system. That you can achieve one with external tools doesn't change that JavaScript will never have a static type system.

In other words, "No", full stop, is the only correct answer to this question.

Reviewer: hmm, yeah, you could argue `Not a _full_ type system` implies it could get some sort of static type system.  and that's not what type annotations would be introducing
